src/modules/flow/http-client/http-client.c

```diff
@@ -986,22 +1000,14 @@ translate_http_method(const char *method)
         sol_str_slice_from_str(method), SOL_HTTP_METHOD_INVALID);
 }
 
-static int
-request_node_get_timeout(int32_t timeout, int *out_timeout)
+static void
+clamp_int_value_if_needed(int *dst, int32_t src)
 {
-    if (timeout > INT_MAX) {
-        SOL_WRN("The timeout is bigger than %d", INT_MAX);
-        *out_timeout = INT_MAX;
-        return 0;
-    }
-
-    if (timeout < 0) {
-        SOL_ERR("Timeout can not be a negative number!");
-        return -EINVAL;
-    }
-
-    *out_timeout = timeout;
-    return 0;
+    if (src > INT_MAX) {
```

Comment:
btw, this function shouldn't exist here. All the API should be int32_t (or uint32_t). Internally, if MHD only supports int, then it should clamp there -- or doesn't care, since MHD is linux-only and requires 32bits or bigger, so int == int32_t.

in any way it should be in the node type.

(BTW reminds me that our sol_timeout_add() need an API fix :-()